### ISSUE TITLE
Request to bump ocamlformat version

### ISSUE THREAD

**Issue by BridgeTheMasterBuilder** [doomemacs/doomemacs#7412]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.
- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.
- [X] The issue can be reproduced on the **latest** available commit of Doom.
- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

It should be able to format an Ocaml file using the latest version of `ocamlformat` without invoking the binary with obsolete options.

### Current behavior

A few months ago `ocamlformat` introduced a breaking change that now makes `ocamlformat-newline-and-indent` trigger an error when using Doom with the latest version of `ocamlformat`.

```
ocamlformat errors:
ocamlformat: unknown option '--numeric', did you mean '-n'?
             unknown option '--range'.
Usage: ocamlformat [OPTION]… [SRC]…
Try 'ocamlformat --help' for more information.
```

This has already been fixed upstream, the package version just needs to be bumped up in Doom. It needs to be bumped up to the latest version, `0.26.0`.

### Steps to reproduce

Format any Ocaml file using ocamlformat version 0.26.0.

### System Information

https://pastebin.com/Lz5rhmZK

**Comment by coveres** [comment]
Is this forgotten? Would be nice to get it fixed.

**Comment by azolus** [comment]
Hi there,
I'm new to doom (it's awesome, thanks for the hard work) and was running into some eerily similar issues with ocamlformat lately:
```
ocamlformat errors:
ocamlformat: unknown option '--numeric', did you mean '-n'?
             unknown option '--range'.
Usage: ocamlformat [OPTION]… [SRC]…
Try 'ocamlformat --help' for more information.
```
This happens each time I hit enter in insert-mode. I assume (since it matches perfectly with the example given above) that the bump would fix this issue. The problem does not stem from `format +onsave`, because formatting the whole file with ocamlformat works fine.

After some experimentation I found the culprit: `ocamlformat-newline-and-indent` (`ocamlformat-region` and `ocamlformat-line` have the same problem).

This error makes editing ocaml code extremely awkward and I had to disable the language feature again :(

I understand that there are many todos concerning doom so I wanted to ask if there is an easy workaround that I missed until a fix is available.
